### PR TITLE
feat: Add more context to support chat

### PR DESCRIPTION
**Pull request by MiloradFilipovic**
## Summary
Adding more details about what user is currently seeing to the support chat request:
```
type UserContext {
	activeNodeInfo?: {
		node?: INode;
		nodeIssues?: INodeIssues;
		nodeInputData?: IDataObject;
		referencedNodes?: NodeExecutionSchema[];
		executionStatus?: {
			status: string;
			error?: ErrorContext['error'];
		};
	};
	activeCredentials?: Pick<ICredentialType, 'name' | 'displayName'>;
	currentView?: VIEWS;
}
```

[Accompanying AI Service PR](https://github.com/n8n-io/ai-assistant-service/pull/167)

## Related Linear tickets, Github issues, and Community forum posts
Closes ADO-2659
Closes ADO-2646

## Review / Merge checklist

- [ ] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) 
- [ ] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created.
- [ ] Tests included. 
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

File: packages/editor-ui/src/components/CredentialEdit/CredentialEdit.vue [hunk 1]
```diff
@@ -420,6 +420,7 @@ async function beforeClose() {
 	}
 
 	if (!keepEditing) {
+		uiStore.activeCredentialType = null;
```
Comment (mutdmour) [inline packages/editor-ui/src/components/CredentialEdit/CredentialEdit.vue:423]: why was this added?

File: packages/editor-ui/src/utils/nodeTypesUtils.ts [hunk 2]
```diff
@@ -581,6 +583,66 @@ export function processNodeForAssistant(node: INode, propsToRemove: string[]): I
...
+		const credentialInUse = node.parameters[authField?.name ?? ''];
+		const availableAuthOptions = getNodeAuthOptions(nodeType);
+		authType = availableAuthOptions.find((option) => option.value === credentialInUse);
+	}
+	let nodeInputData: { inputNodeName?: string; inputData?: IDataObject } | undefined = undefined;
+	const ndvInput = ndvStore.ndvInputData;
+	if (isNodeReferencingInputData(node) && ndvInput?.length) {
+		const inputData = ndvStore.ndvInputData[0].json;
+		const inputNodeName = ndvStore.input.nodeName;
+		nodeInputData = {
```
Comment (mutdmour) [inline packages/editor-ui/src/utils/nodeTypesUtils.ts:609]: what about multiple input nodes?

File: packages/editor-ui/src/components/CredentialEdit/CredentialEdit.vue (hunk 1 above)
Comment (MiloradFilipovic) [reply packages/editor-ui/src/components/CredentialEdit/CredentialEdit.vue:423]: This modal is setting `activeCredentialType` [here](https://github.com/n8n-io/n8n/blob/25a7b9897ffa6b0fe572792ff3d9c581cd018cfc/packages/editor-ui/src/components/CredentialEdit/CredentialEdit.vue#L990) but it's never reset once it's closed.

File: packages/editor-ui/src/utils/nodeTypesUtils.ts (hunk 2 above)
Comment (MiloradFilipovic) [reply packages/editor-ui/src/utils/nodeTypesUtils.ts:609]: This will send the input the user is seeing in ndv. Let me know if you think we should be getting everything but I think this is enough since we are providing user's visual context to the assistant here.

Comment (mutdmour) [reply packages/editor-ui/src/utils/nodeTypesUtils.ts:609]: If that's enough for our usecase then all good

**Review by mutdmour — DISMISSED** [review]

**Review by mutdmour — DISMISSED** [review]

**Review by mutdmour — APPROVED** [review]
Reminder to add more tests

**Comment by janober** [comment]
Got released with [`n8n@1.62.1`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.62.1)
